driver/pelican/types/thermSchedule.go

```diff
@@ -0,0 +1,252 @@
+package types
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/parnurzeal/gorequest"
+	rrule "github.com/teambition/rrule-go"
+)
+
+// Login, Authentication, Thermostat ID Retrieval Structs
+type thermIDRequest struct {
+	Resources []thermIDResources `json:"resources"`
+}
+
+type thermIDResources struct {
+	Children    []thermIDChild `json:"children"`
+	GroupId     string         `json:"groupId"`
+	Permissions string         `json:"permissions"`
+}
+
+type thermIDChild struct {
+	Id          string `json:"id"`
+	Permissions string `json:"permissions"`
+}
+
+// Thermostat Settings Structs
+type settingsRequest struct {
+	Epnum    float64         `json:"epnum"`
+	Id       string          `json:"id"`
+	Nodename string          `json:"nodename"`
+	Userdata settingsWrapper `json:"userdata"`
+}
+
+type settingsWrapper struct {
+	Epnum    float64 `json:"epnum"`
+	Fan      string  `json:"fan"`
+	Nodename string  `json:"nodename"`
+	Repeat   string  `json:"repeat"`
+}
+
+// Thermostat Schedule By Day Decoding Structs
+type scheduleRequest struct {
+	ClientData scheduleSetTimes `json:"clientdata"`
+}
+
+type scheduleSetTimes struct {
+	SetTimes []scheduleTimeBlock `json:"setTimes"`
+}
+
+type scheduleTimeBlock struct {
+	HeatSetting float64 `json:"heatSetting"`
+	CoolSetting float64 `json:"coolSetting"`
+	StartValue  string  `json:"startValue"`
+	System      string  `json:"systemDisplay"`
+}
+
+// Thermostat Schedule Structs
+
+// Struct mapping each day of the week to its daily schedule
+type ThermostatSchedule struct {
+	DaySchedules map[string]ThermostatDaySchedule `msgpack:"day_schedules"`
```

Comment:
See comment above on potentially simplifying the schedule representation.

Reply:
Removed the intermediate struct!